tests/functional/openstackdataplanedeployment_controller_test.go

```diff
@@ -30,23 +38,116 @@ var _ = Describe("Dataplane Deployment Test", func() {
-			DeferCleanup(th.DeleteInstance, CreateDataplaneDeployment(dataplaneDeploymentName, DefaultDataPlaneDeploymentSpec()))
 			CreateSSHSecret(dataplaneSSHSecretName)
+			CreateSSHSecret(neutronOvnMetadataSecretName)
+			CreateSSHSecret(novaNeutronMetadataSecretName)
+			CreateSSHSecret(novaCellComputeConfigSecretName)
+			DeferCleanup(th.DeleteInstance, CreateDataplaneNodeSet(dataplaneNodeSetName, DefaultDataPlaneNodeSetSpec()))
+			DeferCleanup(th.DeleteInstance, CreateDataplaneDeployment(dataplaneDeploymentName, DefaultDataPlaneDeploymentSpec()))
 		})
 
 		It("should have conditions set", func() {
+			Eventually(func(g Gomega) {
+				nodeSet := dataplanev1.OpenStackDataPlaneNodeSet{}
+				baremetal := baremetalv1.OpenStackBaremetalSet{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      nodeSet.Name,
+						Namespace: nodeSet.Namespace,
+					},
+				}
+				// Create config map for OVN service
+				Eventually(func(g Gomega) {
+					ovnConfigMapName := types.NamespacedName{
+						Namespace: namespace,
+						Name:      "ovncontroller-config",
+					}
+					mapData := map[string]interface{}{
+						"ovsdb-config": "test-ovn-config",
+					}
+					th.CreateConfigMap(ovnConfigMapName, mapData)
+				}).Should(Succeed())
+
+				// Set node set deployment ready condition to False
+				Eventually(func(g Gomega) {
+					g.Expect(th.K8sClient.Get(th.Ctx, dataplaneNodeSetName, &nodeSet)).To(Succeed())
+					nodeSet.Status.Conditions.MarkFalse(
+						condition.Type(fmt.Sprintf(dataplanev1.NodeSetServiceDeploymentReadyCondition, nodeSet.Name, "download-cache")),
+						condition.RequestedReason,
+						condition.SeverityInfo,
+						fmt.Sprintf(dataplanev1.NodeSetServiceDeploymentReadyMessage, nodeSet.Name, "download-cache"))
+					g.Expect(th.K8sClient.Status().Update(th.Ctx, &nodeSet)).To(Succeed())
+				}).Should(Succeed())
+
+				// Set baremetal provisioning conditions to True
+				Eventually(func(g Gomega) {
+					g.Expect(th.K8sClient.Get(th.Ctx, dataplaneNodeSetName, &baremetal)).To(Succeed())
+					baremetal.Status.Conditions.MarkTrue(
+						baremetalv1.OpenStackBaremetalSetBmhProvisioningReadyCondition,
+						baremetalv1.OpenStackBaremetalSetBmhProvisioningReadyMessage)
+					g.Expect(th.K8sClient.Status().Update(th.Ctx, &baremetal)).To(Succeed())
+
+				}).Should(Succeed())
+
+				// Create all services necessary for deployment
+				for _, serviceName := range nodeSet.Spec.Services {
+					dataplaneServiceName := types.NamespacedName{
+						Name:      serviceName,
+						Namespace: namespace,
+					}
+					CreateDataplaneService(dataplaneServiceName)
+					service := GetService(dataplaneServiceName)
+					//OVN needs config map resource
+					if serviceName == "ovn" {
+						service.Spec.ConfigMaps = append(service.Spec.ConfigMaps, "ovncontroller-config")
+
+						Eventually(func(g Gomega) {
+							g.Expect(k8sClient.Update(ctx, service)).To(Succeed())
+						}, th.Timeout, th.Interval).Should(Succeed())
+					}
```

Comment:
You don't need to modify the service. It is pre-created by the dataplane-operator automatically based on the yaml files stored under config/services. You only need the ConfigMap to exists.